src/transformers/dynamic_module_utils.py

```diff
@@ -143,9 +145,25 @@ def get_class_in_module(class_name, module_path):
     """
     Import a module on the cache directory for modules and extract a class from it.
     """
-    module_path = module_path.replace(os.path.sep, ".")
-    module = importlib.import_module(module_path)
-    return getattr(module, class_name)
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        module_dir = Path(HF_MODULES_CACHE) / os.path.dirname(module_path)
+        module_file_name = module_path.split(os.path.sep)[-1] + ".py"
+
+        # Copy to a temporary directory. We need to do this in another process to avoid strange and flaky error
+        # `ModuleNotFoundError: No module named 'transformers_modules.[module_dir_name].modeling'`
+        shutil.copy(f"{module_dir}/{module_file_name}", tmp_dir)
+        # On Windows, we need this character `r` before the path argument of `os.remove`
+        cmd = f'import os; os.remove(r"{module_dir}{os.path.sep}{module_file_name}")'
+        subprocess.run(["python", "-c", cmd])
```

Comment:
If something goes wrong in the `subprocess.run`, no error will be thrown in the process that calls this method.
I think we should capture/check the output of `subprocess.run`, and do something: 

- either: not to call `shutil.copyfile` below (although this makes the test flaky in this logic branch)
- or: throw an error manually with some information

Let me know if you have suggestion :-)

Reply:
Why do we need to do something? If there is a problem deleting the file (which we copy just after), at worst we get the flaky failure again (though it should be extremely rare at this stage).

Reply:
yeah, right!